quickwit-cli/src/lib.rs

```
@@ -220,15 +223,40 @@ pub async fn index_data_cli(args: IndexDataArgs) -> anyhow::Result<()> {
     Ok(())
 }
 
-fn create_source_config_from_args(input_path_opt: Option<PathBuf>) -> SourceConfig {
-    let params = FileSourceParams {
-        filepath: input_path_opt,
-    };
-    SourceConfig {
-        id: "cli_source".to_string(),
-        source_type: "file".to_string(),
-        params: serde_json::to_value(params).unwrap(),
+async fn create_source_config_from_args(
+    source_config_path_opt: Option<&PathBuf>,
+    input_path_opt: Option<&PathBuf>,
+) -> anyhow::Result<SourceConfig> {
+    if let Some(source_config_path) = source_config_path_opt {
```

We should return an error if both Option are "Some"